backends/cassandra_test.go

```diff
@@ -0,0 +1,205 @@
+package backends
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/gocql/gocql"
+	"github.com/prebid/prebid-cache/utils"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCassandraClientGet(t *testing.T) {
+	cassandraBackend := &CassandraBackend{}
+
+	type testInput struct {
+		cassandraClient CassandraDB
+		key             string
+	}
+
+	type testExpectedValues struct {
+		value string
+		err   error
+	}
+
+	testCases := []struct {
+		desc     string
+		in       testInput
+		expected testExpectedValues
+	}{
+		{
+			"CassandraBackend.Get() throws a Cassandra ErrNotFound error",
+			testInput{
+				&errorProneCassandraClient{errorToThrow: gocql.ErrNotFound},
+				"someKeyThatWontBeFound",
+			},
+			testExpectedValues{
+				value: "",
+				err:   utils.KeyNotFoundError{},
+			},
+		},
+		{
+			"CassandraBackend.Get() throws an error different from Cassandra ErrNotFound error",
+			testInput{
+				&errorProneCassandraClient{errorToThrow: errors.New("some other get error")},
+				"someKey",
+			},
+			testExpectedValues{
+				value: "",
+				err:   errors.New("some other get error"),
+			},
+		},
+		{
+			"CassandraBackend.Get() doesn't throw an error",
+			testInput{
+				&goodCassandraClient{key: "defaultKey", value: "aValue"},
+				"defaultKey",
+			},
+			testExpectedValues{
+				value: "aValue",
+				err:   nil,
+			},
+		},
+	}
+
+	for _, tt := range testCases {
+		cassandraBackend.client = tt.in.cassandraClient
+
+		// Run test
+		actualValue, actualErr := cassandraBackend.Get(context.TODO(), tt.in.key)
+
+		// Assertions
+		assert.Equal(t, tt.expected.value, actualValue, tt.desc)
+		assert.Equal(t, tt.expected.err, actualErr, tt.desc)
+	}
+}
+
+func TestCassandraClientPut(t *testing.T) {
+	cassandraBackend := &CassandraBackend{
+		defaultTTL: 50,
+	}
+
+	type testInput struct {
+		cassandraClient CassandraDB
+		key             string
+		valueToStore    string
+		ttl             int
+	}
+
+	type testExpectedValues struct {
+		value string
+		err   error
+	}
+
+	testCases := []struct {
+		desc     string
+		in       testInput
+		expected testExpectedValues
+	}{
+		{
+			"CassandraBackend.Put() throws error",
+			testInput{
+				&errorProneCassandraClient{errorToThrow: gocql.ErrNoConnections},
+				"someKey",
+				"someValue",
+				10,
```

Comment:
Nitpick: I don't understand that these values are for. Could you please add named initializers in the tests for readability? ex:
```
testInput{
	cassandraClient: &errorProneCassandraClient{errorToThrow: gocql.ErrNoConnections},
	key:             "someKey",
	valueToStore:    "someValue",
	ttl:             10,
}
```